added More Articles component

Thindatuo opened the pull request:
**This pull request update a component which forms the "more article" section.** 

**Component Description:** 
the component consists of a title "More article", two images, and two buttons and two blocks of text under each of them. Every element inside this component are user defined, including the title and button description text. Also, the present of the component will automatically self-adjust according to the screen size. 

**Component look (without inserting two images):**
big screen:
<img width="875" alt="image" src="https://user-images.githubusercontent.com/122959510/230500774-91d58858-91b6-427f-acea-5964b411d656.png">
phone screen:
<img width="527" alt="image" src="https://user-images.githubusercontent.com/122959510/230500847-9602e470-c784-4061-b3ef-62c81a8113e6.png">
dark mode:
<img width="511" alt="image" src="https://user-images.githubusercontent.com/122959510/230501022-299a42da-02f7-47b8-9251-6ef6cf10e598.png">
<img width="881" alt="image" src="https://user-images.githubusercontent.com/122959510/230501113-e5b01595-b787-47f3-b7cc-1db5cc6ef0a2.png">

Also, unit test is included into the fold, which covers 100% of the code, and the coverage looks like below:
![image](https://user-images.githubusercontent.com/122959510/232894730-0d1c3753-2589-4e18-bc8b-d5b0ab3eda07.png)


**File Changed (added):**:
MoreArticles.jsx
MoreArticlesOptions.jsx
MoreArticlesComponent.jsx
MoreArticlesPreview.png
registeredComponent.js
MoreArticles.test.jsx

ebacolod reviewed (changes requested):
I've made comments on specific lines of code in my review. They'll point out where I believe some issues are and provide recommendations on how to fix them.  Here's a summary of the issues I've found:

- Input cannot be changed
- Responsiveness bugs on small and very large screens
- Options are difficult to read
- Full gradient is not being used
- Test file is not included

Thindatuo commented:
Thank you for the detailed review and suggestions! All the suggested changes are made int he latest commit to this PR.

ebacolod reviewed (approved):
Thanks for implementing a lot of the new changes. These changes make your component work well and look cleaner. Also, I ran your tests and got full coverage. 

<img width="954" alt="image" src="https://user-images.githubusercontent.com/58874823/232375433-1c4457c5-026a-4df4-95d7-479e6388c770.png">

The last thing you should do is update the original comment on this PR talking about the test file and coverage being added: 


<img width="680" alt="image" src="https://user-images.githubusercontent.com/58874823/232375857-27a40735-a24c-4f92-915f-8c497217c10e.png">

This request doesn't change the code, so I'll approve this PR. Nice work!

Lettuce05 commented:
Had to close and reopen pull request in order to run GitHub action. Action would not run on preexisting pull requests.

carruthjhall reviewed (changes requested):
+The component is responsive for all screen sizes, and looks correct in light and dark modes.
+The options are able to be edited and are reflected on the preview.
+Able to get full coverage on testing

-The placeholder images don't load . You can fix this by 
1) Downloading the pngs from Figma
2) Uploading them to the our group image repo [cs472-team6/cs472-team6-resources](https://github.com/cs472-team6/cs472-team6-resources/tree/main/images) 
3) Go to the now uploaded image and copy the image link address and then use that as the URL in your component

-Maybe change "Artilce left/top" to "Left Article" and "Article right/top" to "Right Article" just for

Once these fixes are applied then the component will be perfect. Good work!

carruthjhall commented on src/components/MoreArticles/MoreArticles.jsx:
```diff
@@ -0,0 +1,57 @@
+                        <img src={imgLink1} className="w-[300px] h-[192px] md:w-[361px] 
+                             md:h-[231.19px] lg:w-[436px] lg:h-[280px] mb-4" alt="ArticleImage"></img>
+                        <div className="mb-6">
+                            <button className="w-[79px] h-[40px] md:w-[93px] md:h-[44px] rounded-[40px] bg-black 
+                                dark:bg-white mr-4 text-white dark:text-black font-bold" href={buttonLink1}>
+                                {buttonText1}
+                            </button>
```
-Wrap each of the buttons like so:
`<a href={buttonLink}>`
`    <button>`
`    </button>`
`</a>`
so that the buttons will go to the links, because right now they aren't working.

-Add "https://" to the beginning of your links
-There's a typo in google.com

carruthjhall commented on src/components/MoreArticles/MoreArticles.jsx:
```diff
@@ -0,0 +1,57 @@
+                                {buttonText2}
+                            </button>
+                        </div>
+                        <div className="w-[300px] h-[78px] md:w-[362px] md:h-[80px] lg:w-[360px] lg:h-[84px]"> 
+                            <div className="text-[26px] md:text-[32px] font-bold text-black dark:text-white leading-8">{text2}</div>
+                        </div>
+                        <div className="text-[18px] font-Inter font-bold text-black dark:text-white">{author}</div>
+                    </div>
```
-For both article sections, moving the `{author}` div into the div containing the `{text2}` div will fix the description overflowing over the author's name when having a lengthy description.
-Make the leading-8 to leading-10 so that there's a little more space in the description.

Thindatuo commented:
> +The component is responsive for all screen sizes, and looks correct in light and dark modes. +The options are able to be edited and are reflected on the preview. +Able to get full coverage on testing
> 
> -The placeholder images don't load . You can fix this by
> 
> 1. Downloading the pngs from Figma
> 2. Uploading them to the our group image repo [cs472-team6/cs472-team6-resources](https://github.com/cs472-team6/cs472-team6-resources/tree/main/images)
> 3. Go to the now uploaded image and copy the image link address and then use that as the URL in your component
> 
> -Maybe change "Artilce left/top" to "Left Article" and "Article right/top" to "Right Article" just for
> 
> Once these fixes are applied then the component will be perfect. Good work!

Thanks for the details review and the suggestions! Changes have been made accordingly.

carruthjhall reviewed (changes requested):
Great fixes. Everything that has been addressed, has been fixed.

The only thing that needs to be fixed is fixing the faqSection bug so that the test bot can pass. 
You can fix by changing the lines in:
faqComponent.jsx: Change `./FAQSection` to `./faqSection` in `import faqSection from "./FAQSection";`
faqComponent.test.jsx: Change `./FAQSection` to `./faqSection` in `import faqSection from "./FAQSection";`

or you can sync your fork on this branch.

Thindatuo commented:
> Great fixes. Everything that has been addressed, has been fixed.
> 
> The only thing that needs to be fixed is fixing the faqSection bug so that the test bot can pass. You can fix by changing the lines in: faqComponent.jsx: Change `./FAQSection` to `./faqSection` in `import faqSection from "./FAQSection";` faqComponent.test.jsx: Change `./FAQSection` to `./faqSection` in `import faqSection from "./FAQSection";`
> 
> or you can sync your fork on this branch.

Thank you! I get your point, and I thought that though the fap component is broken on my branch, when merging my branch to main, it only adds what I've added/changed, so if the faq component in the current main is fine, it would ignore the broken version on my branch and only look for what I've changed. If it's not the case, I would make change to it.

carruthjhall reviewed (approved):
Everything works perfectly and the tests were able to pass now. I approve this PR. Great work again!
